apache/trafficcontrol
# TP: adds the ability to rearrange table columns as well as toggle column visibility

This PR adds the ability to show/hide table columns in TP as well as reorder the visible columns of a table. The changes made by the user will be persisted in the browsers local storage.

I tackled an easy table (the types table) to start and prove out the concept.


## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?
1. Run the [UI tests](https://github.com/apache/trafficcontrol/tree/master/traffic_portal/test/end_to_end)
2. Manually verify
- Launch TP
- navigate to Configure > Types
- reorder columns via drag/drop on the column header
- show/hide columns by using the following drop down

![image](https://user-images.githubusercontent.com/251272/59782928-40118200-927c-11e9-9a37-501b7302f542.png)

Note: I'm not going to document this functionality or add it to the changelog until it is added to other tables in subsequent PRs.

## The following criteria are ALL met by this PR
